conformance/reports/v1.1.0/istio-istio/1.22-report.yaml

```diff
@@ -0,0 +1,110 @@
+apiVersion: gateway.networking.k8s.io/v1alpha1
+date: "2024-04-25T14:54:49Z"
+gatewayAPIChannel: experimental
+gatewayAPIVersion: v1.1.0-rc1
+implementation:
+  contact:
+  - '@istio/maintainers'
+  organization: istio
+  project: istio
+  url: https://istio.io/
+  version: "1.22"
+kind: ConformanceReport
+mode: default
+profiles:
+- core:
+    result: success
+    statistics:
+      Failed: 0
+      Passed: 12
+      Skipped: 0
+  extended:
+    result: success
+    statistics:
+      Failed: 0
+      Passed: 1
+      Skipped: 0
+    supportedFeatures:
+    - GatewayPort8080
+    - GatewayStaticAddresses
+  name: GATEWAY-GRPC
+  summary: Core tests succeeded. Extended tests succeeded.
+- core:
+    result: success
+    statistics:
+      Failed: 0
+      Passed: 33
+      Skipped: 0
+  extended:
+    result: success
+    statistics:
+      Failed: 0
+      Passed: 18
+      Skipped: 0
+    supportedFeatures:
+    - GatewayPort8080
+    - GatewayStaticAddresses
+    - HTTPRouteBackendRequestHeaderModification
+    - HTTPRouteBackendTimeout
+    - HTTPRouteHostRewrite
+    - HTTPRouteMethodMatching
+    - HTTPRouteParentRefPort
+    - HTTPRoutePathRedirect
+    - HTTPRoutePathRewrite
+    - HTTPRoutePortRedirect
+    - HTTPRouteQueryParamMatching
+    - HTTPRouteRequestMirror
+    - HTTPRouteRequestMultipleMirrors
+    - HTTPRouteRequestTimeout
+    - HTTPRouteResponseHeaderModification
+    - HTTPRouteSchemeRedirect
+  name: GATEWAY-HTTP
+  summary: Core tests succeeded. Extended tests succeeded.
+- core:
+    result: success
+    statistics:
+      Failed: 0
+      Passed: 11
+      Skipped: 0
+  extended:
+    result: success
+    statistics:
+      Failed: 0
+      Passed: 1
+      Skipped: 0
+    supportedFeatures:
+    - GatewayPort8080
+    - GatewayStaticAddresses
+  name: GATEWAY-TLS
+  summary: Core tests succeeded. Extended tests succeeded.
+- core:
+    result: success
+    statistics:
+      Failed: 0
+      Passed: 2
+      Skipped: 0
+  extended:
+    result: partial
+    skippedTests:
+    - MeshFrontendHostname
```

Comment:
Why is this test being skipped? (x-ref https://github.com/kubernetes-sigs/gateway-api/pull/3021#discussion_r1580020525)

Reply:
We do not pass it

Reply:
We don't "FAIL" it because we run our tests in CI/CD. Fail means unexpected failure, skipped means "known failure"

Reply:
Is this something that will eventually pass on Istio or is it just incompatible?

Reply:
It can but it requires substantial re-architecture. We are very strongly considering it.

Also ambient mode will pass it. I guess I should probably make 2 entries here, one for ambient? 

Ambient fails MeshConsumerRoute though. Again, its fixable but will need a ton of work that is planned in the ~9 month time horizon

Reply:
In this case, the test your skipping is testing an extended feature, maybe that feature should not show up as supported unless all tests that include it pass? Not entirely sure how to handle this, and also does not need to block this PR, just trying to get some context for a follow up issue.

/cc @mlavacca @shaneutt

Reply:
There is no feature for 'MeshFrontendHostname' though... I agree its a bit weird though

Reply:
We covered the case of having skipped tests in our conformance reports rulebook:
https://github.com/kubernetes-sigs/gateway-api/blob/7320257302905254dcebf4f5c424741cf1556c06/conformance/reports/README.md#L123-L127

If we'd strictly adhere to the rules written therein, the extended feature should be removed, hence +1 on the theoretical argument brought up by Rob
> we should remove that from the list of supported features in this report until all tests with that listed as a supported feature pass

I'm not very familiar with the feature at issue, but we could think of relaxing the `SupportHTTPRouteResponseHeaderModification` requirements by introducing a specific feature to use for the 
`MeshFrontendHostname` test and m maybe others, if it makes sense.

Reply:
Yeah I think the problem is `MeshFrontendHostname` is a test that represents feature `SupportHTTPRouteResponseHeaderModification`. 

But it really has nothing to do with `SupportHTTPRouteResponseHeaderModification`. We support header modification just fine, we just don't support `MeshFrontendHostname` which is not a "feature"

Reply:
Also I don't really get it. The doc says you have to pass 100% of tests in order to list a report? Am I interpreting that correctly that any implementation that is not 100% conformant cannot even list a report at all that accurate states "We passed all of these tests, but not X, Y, Z"?

If so, what is the point of the report -- it could be reduced to `passed` (which probably means I misintepret it :slightly_smiling_face: )

Reply:
This feature was introduced in the `MeshFrontendHostname` test by @kflynn in this PR: https://github.com/kubernetes-sigs/gateway-api/pull/2312. Specifically, the PR description states:

> The MeshFrontendHostname test actually relies on a header modification, but wasn't tagged as such.

I am not opposed to adding a new `MeshFrontendHostname` feature (given that I do not see any problem in leaving the `SupportHTTPRouteResponseHeaderModification`, as I think it makes sense), but I'd like to double-check with @robscott and @kflynn as well.

Reply:
> Also I don't really get it. The doc says you have to pass 100% of tests in order to list a report? Am I interpreting that correctly that any implementation that is not 100% conformant cannot even list a report at all that accurate states "We passed all of these tests, but not X, Y, Z"?
> 
> If so, what is the point of the report -- it could be reduced to `passed` (which probably means I misintepret it 🙂 )

Yes, according to the rules we have, a report can be accepted only in case all the results are `success`.
This means:
- 100% conformance for core tests
- 100% conformance for all the tests related to the supportedFeatures (such as the `SupportHTTPRouteResponseHeaderModification` here)

It makes sense to have a complex report because this is firstly intended for implementations to understand what's going on in their project; you as an implementation are interested in having a complete overview of the testing outcome and this is what the report is for. The submission to this repo is only the last step, and should be allowed only to implementations complying with specific requirements (btw, these requirements have been added recently and are enforced only on the new profiles).

Reply:
One thing to clarify - we have plenty of tests that cover the intersection of features. For example:

https://github.com/kubernetes-sigs/gateway-api/blob/7320257302905254dcebf4f5c424741cf1556c06/conformance/tests/httproute-redirect-port-and-scheme.go#L43-L44

An implementation could still be considered to support one of those features even if it doesn't support the other. That means we likely need to clean up the "as well as all the tests related to the supported extended features" wording in the doc @mlavacca linked to above.

Reply:
Yep, I agree the wording here is not sufficient to express the corner case where `testA` requires `Feature1` and `Feature2`, and the implementation supports `Feature1` only.
According to the wording here, `testA` should be successful, but we know it should not. I think we can fix it when addressing https://github.com/kubernetes-sigs/gateway-api/issues/3036, as it will 100% require changes in that paragraph.

Reply:
This is no longer skipped with the latest